### PR TITLE
check last timber is saved

### PR DESCRIPTION
**Pull request by druiz0992**
## What does this implement/fix? Explain your changes.
This PR fixes a bug during block generation. It is possible that if there are many transactions in the previous block, the timber is not saved by the time a new block is being built. This PR detects this case and builds a new block based on expected timber rather than on actual timber.

## Does this close any currently open issues?
No

### CODE REVIEW

File: nightfall-optimist/src/classes/block.mjs [hunk 1]
```diff
@@ -93,13 +94,15 @@ class Block {
       await getLatestBlockInfo();
     let blockNumberL2 = dbPrevBlockNumberL2 + 1; // We increment it as this is what the next block should be
     let previousBlockHash = dbBlockHash;
-    // It's possible that the previously made block hasn't been added to the blockchain yet.
-    // In that case, this block will have the same block number as the previous block
+    const timberBlockNumberL2 = await getNumberOfL2Blocks();
```
Comment (pawelgrzybek) [inline nightfall-optimist/src/classes/block.mjs:97]: This should be fetched simultaneously with `getLatestBlockInfo` a few lines above. Like so:

```js
const [{ blockNumberL2: dbPrevBlockNumberL2, blockHash: dbBlockHash }, timberBlockNumberL2] =
      await Promise.all([getLatestBlockInfo(), getNumberOfL2Blocks()]);
```

**Review by pawelgrzybek — CHANGES_REQUESTED** [review]

File: nightfall-optimist/src/classes/block.mjs (hunk 1 above)
Comment (druiz0992) [reply nightfall-optimist/src/classes/block.mjs:97]: It’s not needed. There are read from different collections, updated at different times. So, there is no guarantee that data is more in sync. But I can do the change if you insist

Comment (pawelgrzybek) [reply nightfall-optimist/src/classes/block.mjs:97]: I dont insist. I assume there are two individual calls and the second one is not dependant one the 1st one. Keep it as it is then.

Comment (druiz0992) [reply nightfall-optimist/src/classes/block.mjs:97]: looks better your way so i made the change

**Review by pawelgrzybek — APPROVED** [review]

**Review by israelboudoux — APPROVED** [review]

**Pull request merged**